### PR TITLE
Update Engine - Node

### PR DESCRIPTION
**Pull request by donavanbecker**
This PR is to update the node engine to be in line with ```hap-nodejs``` and their node engine.

### CODE REVIEW

**Comment by donavanbecker** [comment]
> This is good, but wondering if we should set the minimum to be the current nodeJS LTS version?
> 
would we want to do ```latest``` or ```stable```

**Comment by donavanbecker** [comment]
@NorthernMan54 I have updated it to be ```Node.js Version | v12.16.1```

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "homebridge": "bin/homebridge"
   },
   "engines": {
-    "node": ">=10.17.0"
+    "node": ">=12.16.1"
```
Comment (donavanbecker) [inline package.json]: @NorthernMan54 I have updated it to be ```Node.js Version | v12.16.1```

**Comment by oznu** [comment]
Set the minimum in engines to v8.15.1. Everything still works on this version.

Update the readme to say v10+ as that's a "Current" LTS still.

**Comment by oznu** [comment]
Forcing people to update Node.js more often than they have to will cause a spike in issues.

**Comment by donavanbecker** [comment]
> Set the minimum in engines to v8.15.1. Everything still works on this version.
> 
> Update the readme to say v10+ as that's a "Current" LTS still.

@oznu I have updated the readme to say v10 and also changed the engine to v18.15.1.

**Pull request merged**